src/Common/Telemetry/Telemetry.php

```diff
@@ -639,4 +653,115 @@ public static function disable_modal( $slug, $enable = false ) {
 		$option_slug = Config::get_container()->get( Opt_In_Template::class )->get_option_name( $slug );
 		update_option( $option_slug, $enable );
 	}
+
+	/**
+	 * Filters the data arguments for TEC plugins.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,mixed> $args The array of data arguments to filter.
+	 *
+	 * @return array<string,mixed> $args The array of filtered data arguments.
+	 */
+	public function filter_data_args( $args ): array {
+		// We only want to mess with the data block.
+		$telemetry = json_decode( $args['telemetry'], true );
+
+		list( $changed, $telemetry ) = $this->add_licenses( $telemetry );
+
+		if ( tribe_is_truthy( $changed ) ) {
+			$args['telemetry'] = json_encode( $telemetry );
+		}
+
+		/**
+		 * Allows overriding the data arguments for TEC plugins.
+		 *
+		 * @since TBD
+		 *
+		 * @param array<string,mixed> $args The data arguments to filter.
+		 *
+		 * @return array<string,mixed> $args The filtered data arguments.
+		 */
+		return apply_filters( 'tec_telemetry_data_arguments', $args );
+	}
+
+	/**
+	 * Add premium plugin licenses to the telemetry data.
+	 *
+	 * @since TBD
+	 *
+	 * @param object $telemetry The telemetry data object.
+	 *
+	 * @return array<bool,object> and array in the format:
+	 * 		[
+	 * 			bool $changed   If the data has changed.
+	 * 			obj  $telemetry The (potentially modified) telemetry data object.
+	 * 		]
+	 */
+	protected function add_licenses( $telemetry ) {
+		// No data sent, bail safely.
+		if ( empty( $telemetry ) ){
+			return [ false, $telemetry ];
+		}
+
+		// We don't need the path info for this.
+		$tec_slugs = array_keys( self::get_tec_telemetry_slugs() );
+		// Remove parent slugs - they don't have licenses.
+		$tec_slugs = array_diff( $tec_slugs, self::$base_parent_slugs );
+
+		// Nothing to add, bail safely.
+		if ( empty( $tec_slugs ) ){
+			return [ false, $telemetry ];
+		}
+
+		if ( ! isset( $telemetry['stellar_licenses'] ) ) {
+			// Set up if it doesn't exist.
+			$telemetry['stellar_licenses'] = [];
+		} elseif( ! is_array( $telemetry['stellar_licenses'] ) ) {
+			// Make sure it's an array if it does exist.
+			$telemetry['stellar_licenses'] = (array) $telemetry['stellar_licenses'];
+		}
+
+		// Grab all the options, cached.
+		$options = wp_load_alloptions();
+
+		// Filter out everything but our license keys.
+		$options = array_filter(
+			$options,
+			function( $key ) {
+				return str_starts_with( $key, "pue_install_key_");
```

Comment:
Note this does not get licenses from ET+ and some other plugins as they store them differently. WIP.